### DISCUSSION TITLE
Block-like events

### DISCUSSION THREAD

**Discussion by pdpino** [hoangnm/react-native-week-view#220] [Show and tell]
I'm working on the feature _block-like events_. First working version:

![image](https://user-images.githubusercontent.com/20805451/174190084-59056f4b-6943-4b8b-8195-64801ae30c05.png) ![image](https://user-images.githubusercontent.com/20805451/174189922-cdcbeb1c-cf38-4c08-a7d0-ed85c0b65c02.png)

"Event 5" is a regular event, while "Block" and "Other block" are _block-like_ events, that span multiple days.

The way I see it, this will ~~solve~~ help solving:

* Events that span through multiple days, e.g. https://github.com/hoangnm/react-native-week-view/issues/217#issuecomment-1155011520
* Timeslots, e.g. https://github.com/hoangnm/react-native-week-view/issues/92#issuecomment-847198383, #235 
* Business hours, e.g. #140

(Possible) API change:
```js
const awesomeEvents = [{
    id: 1,
    color: 'lightblue',
    description: 'Block',

    // New field:
    eventType: 'block',

    // Dates indicate block corners:
    startDate: new Date(2022, 5, 17, 9, 30), // left-top corner
    endDate: new Date(2022, 5, 18, 13, 30), // right-bottom corner
}]
```

**Comment by hoangnm** [comment]
I understand the first use case, the event that can span multiple days. However, for the two remaining use cases, how this can help?

**Reply by pdpino** [reply]
For those I'm thinking of adding customization **per event**, such as:
* styling (container border, margins, etc) (e.g. https://github.com/hoangnm/react-native-week-view/issues/228, https://github.com/hoangnm/react-native-week-view/issues/167#issue-1162562216)
* interactions, e.g. enable/disable dragging (https://github.com/hoangnm/react-native-week-view/issues/166), or others

```js
const awesomeEvents = [
  {
    // e.g. business hours
    id: 1,
    eventType: 'block',
    startDate: Monday 8am,
    endDate: Friday 5pm,
    color: 'gray',
    borderColor: 'transparent',
    horizontalPadding: 0,
    disableDrag: true,
    disablePress: true,
    disableLongPress: true,
  },
  // ... regular events
]
```

(I updated the original description, we still need more stuff to solve those use cases :sweat_smile:)

**Comment by pdpino** [comment]
@hoangnm update: the block-like events implementation is ready

(fully working in this branch: https://github.com/pdpino/react-native-week-view/tree/unstable/blocks)

I will split the changes into sequential PRs:
- [x] #229 (ongoing already)
- [x] Make `Event` a `PureComponent` (#252)
- [x] Support stacking events (#255)
- [ ] Support block-like events (#267)

**Comment by rdewolff** [comment]
How is this going to look like when u stack event on top of that grid event ?

**Reply by pdpino** [reply]
By default, block events are behind other events. In the future I'm thinking you could provide a z-index per event (or similar)

(I don't have a screenshot right now, could show later if needed)

**Reply by rdewolff** [reply]
That is amazing, I would love to have some kind of a layer functionality

**Comment by TheTushar2696** [comment]
Hi @pdpino 
Eagerly waiting for this, any update on this.

Thanks in advance
